core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -916,8 +1040,13 @@ private[spark] class ExecutorAllocationManager(
       attempts.count(attempt => unschedulableTaskSets.contains(attempt))
     }
 
+    /**
+     * Will account for only pending tasks and speculative tasks of a current stage.
+     * Will also account for pending tasks from other stages if enabled for streaming use case.
+     * i.e if `spark.dynamicAllocation.streaming.enabled` is enabled.
+     */
     def hasPendingTasks: Boolean = {
-      hasPendingSpeculativeTasks || hasPendingRegularTasks
+      hasPendingSpeculativeTasks || hasPendingRegularTasks || hasPendingTasksFromOtherStagesOfTheJob
```

Comment:
Can you only check "hasPendingTasksFromOtherStagesOfTheJob" if "streamingDRAFeatureEnabled" is enabled?

Reply:
`hasPendingTasksFromOtherStagesOfTheJob` will always be `false` when `streamingDRAFeatureEnabled` is not enabled. 

The underlying data structure `stageIdToNumTasks` will always be empty as update operations on it are always behind `streamingDRAFeatureEnabled` flag.